### PR TITLE
Add Support to YourLand Notebook based on RK3588

### PR DESCRIPTION
**Pull request by gedulab**
Initial support of YourLand notebook based on RK3588.

### CODE REVIEW

**Comment by mariobalanica** [comment]
There's a lot of code duplication. The ACPI code looks mostly unnecessary. 
What issues are you seeing with SD/eMMC?

**Comment by gedulab** [comment]
Sorry for the duplication. Please feel free to remove all the asl files.

**Comment by mariobalanica** [comment]
It's not just the asl files. For instance, RockchipSdhci and ecu under RockchipPlatformLib are unused. There are additional definitions in the DSC & FDF that are not found in other platforms. 

Please clean up the patch and push a new commit.

If you wish to add new features (e.g. Capsule, Ext4) or fix some issues (e.g. with SDHCI), we can discuss these separately.

I see some battery and audio bits in ACPI. Do they work in any OS?

File: edk2-rockchip/Platform/GEDU/YourLand/YourLand.dsc [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  #
+  DEFINE RK_RTC8563_ENABLE = TRUE 
+
+  #
+  # YourLand uses 2 RK806s, no RK860 
+  #
+  DEFINE RK_860X_REGULATOR_ENABLE = FALSE
+
+  #
+  # the drivers in current upstream don't work well
```
Comment (MollySophia) [inline edk2-rockchip/Platform/GEDU/YourLand/YourLand.dsc:45]: May I ask what's the behavior with the current drivers?

File: edk2-rockchip/Platform/GEDU/YourLand/Library/RockchipPlatformLib/RockchipPlatformLib.c [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+    MicroSecondDelay(1000);
+    GpioPinWrite (3, GPIO_PIN_PD3, 0);
+  }  
+  if(KeepOn)
+    GpioPinWrite (3, GPIO_PIN_PD3, 1);
+
+  GpioPinWrite (3, GPIO_PIN_PD4, 1);
+}
+
+static struct regulator_init_data rk806_init_data[] = {
```
Comment (MollySophia) [inline edk2-rockchip/Platform/GEDU/YourLand/Library/RockchipPlatformLib/RockchipPlatformLib.c:40]: I can't see the necessity to reserve this

File: edk2-rockchip/Platform/GEDU/YourLand/Library/RockchipPlatformLib/RockchipPlatformLib.c [hunk 3]
```diff
@@ -0,0 +1,601 @@
...
+
+	RK8XX_VOLTAGE_INIT(SLAVER_PLDO1, 1800000),
+	RK8XX_VOLTAGE_INIT(SLAVER_PLDO2, 1800000),
+	RK8XX_VOLTAGE_INIT(SLAVER_PLDO3, 1800000),
+	RK8XX_VOLTAGE_INIT(SLAVER_PLDO4, 3300000),
+	RK8XX_VOLTAGE_INIT(SLAVER_PLDO5, 2800000),
+	RK8XX_VOLTAGE_INIT(SLAVER_PLDO6, 1800000),
+};
+
+BOOLEAN UlIsUlan()
```
Comment (MollySophia) [inline edk2-rockchip/Platform/GEDU/YourLand/Library/RockchipPlatformLib/RockchipPlatformLib.c:159]: The RockchipPlatformLib.c is already platform-specific. There's no need to take the single-rk806 case into account here.

File: edk2-rockchip/Platform/GEDU/YourLand/Library/RockchipPlatformLib/RockchipPlatformLib.c [hunk 4]
```diff
@@ -0,0 +1,601 @@
...
+}
+
+VOID
+EFIAPI
+PcieIoInit (
+  UINT32 Segment
+  )
+{
+  MmioWrite32(0xFEC4000c, 0x80008); /* output */
+#if 0
```
Comment (MollySophia) [inline edk2-rockchip/Platform/GEDU/YourLand/Library/RockchipPlatformLib/RockchipPlatformLib.c:351]: These can be removed obviously.

File: edk2-rockchip/Platform/GEDU/YourLand/Library/RockchipPlatformLib/RockchipPlatformLib.c (hunk 3 above)
Comment (gedulab) [reply edk2-rockchip/Platform/GEDU/YourLand/Library/RockchipPlatformLib/RockchipPlatformLib.c:159]: you are right, we will fix the issues. many thanks for your review

**Comment by mariobalanica** [comment]
Hi, any update on this?

**Pull request closed without merging**